src/Button/WarningButton.tsx

```diff
@@ -0,0 +1,25 @@
+import styled, { CSSObject } from "styled-components";
+import { darken } from "polished";
+import Button, { ButtonProps } from "./Button";
+
+const WarningButton: React.FC<any> = styled(Button)(
```

Comment:
I think we should use a generic passed to the styled function here instead 
```suggestion
const WarningButton = styled(Button)<ButtonProps>(
```